integration-libs/cdc/components/gigya-raas/gigya-raas.guard.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, inject } from '@angular/core';
+import { CanActivate, UrlTree } from '@angular/router';
+import { GigyaRaasComponentData } from '@spartacus/cdc/core';
+import {
+  AuthGuard,
+  CmsService,
+  NotAuthGuard,
+  PageContext,
+  RoutingService,
+  isNotUndefined,
+} from '@spartacus/core';
+import { Observable, catchError, filter, of, switchMap, take } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GigyaRaasGuard implements CanActivate {
+  protected routingService = inject(RoutingService);
+  protected cmsService = inject(CmsService);
+  protected authGuard = inject(AuthGuard);
+  protected notAuthGuard = inject(NotAuthGuard);
+
+  canActivate(): Observable<boolean | UrlTree> {
+    return this.getComponentData().pipe(
+      switchMap((componentData) => {
+        if (Object.keys(componentData).length === 0) {
+          return of(false);
```

Comment:
@anjana-bl This one is causing a breaking change, as the guards prevent navigation when there’s no componentData for a given component. It’s quite tedious to debug and understand from the perspective of a project that relies on libraries (in our case one page suddenly stopped to work after upgrade, because of this)